src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -0,0 +1,255 @@
+# This file includes functions that are common for all AbstractIdealGlmMhdMultiIonEquations
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+have_nonconservative_terms(::AbstractIdealGlmMhdMultiIonEquations) = True()
+
+function varnames(::typeof(cons2cons), equations::AbstractIdealGlmMhdMultiIonEquations)
+    cons = ("B1", "B2", "B3")
+    for i in eachcomponent(equations)
+        cons = (cons...,
+                tuple("rho_" * string(i), "rho_v1_" * string(i), "rho_v2_" * string(i),
+                      "rho_v3_" * string(i), "rho_e_" * string(i))...)
+    end
+    cons = (cons..., "psi")
```

Comment:
Is this correct? Do even 1D and 2D have the `v3` component?

Reply:
And if yes, is there even a difference between 1D, 2D, and 3D? That is, are the physical and numerical fluxes even different?

(sorry, should have picked this up earlier)

Reply:
Yes, 1D, 2D and 3D have three components for the velocity and for the magnetic field, as in single-fluid MHD. The main difference for TreeMesh is that 1D equations only have fluxes and non-conservative terms in `orientation == 1`, and higher dimensions have additional orientations. Another more subtle difference is that some terms vanish in 1D.

Reply:
OK - thanks for the clarification. It might make sense to add this information in the docs somewhere once you start adding this for other dimensions as well.